src/edu/csus/ecs/pc2/list/ListUtilities.java

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 1989-2023 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.list;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.csus.ecs.pc2.core.LanguageUtilities;
+import edu.csus.ecs.pc2.core.model.IInternalContest;
+import edu.csus.ecs.pc2.core.model.Language;
+import edu.csus.ecs.pc2.core.model.Problem;
+import edu.csus.ecs.pc2.imports.ccs.IContestLoader;
+import edu.csus.ecs.pc2.ui.SubmissionSolutionList;
+import edu.csus.ecs.pc2.validator.clicsValidator.ClicsValidator;
+
+/**
+ * A set of utilities that operate on lists.
+ * 
+ * @author Douglas A. Lane <pc2@ecs.csus.edu>
+ *
+ */
+public class ListUtilities {
+
+	/**
+     * Returns list of files that match the SubmissionSolutionList (judging types location/list)
+     * 
+     * @param files list of files to be filtered
+     * @param submissionList list of SubmissionSampleLocation
```

Comment:
The formal parameter name ("submissionList") in this `@param` statement does not match the actual method header.  I know it's trivial, but please update the documentation to match the code (or vice-versa).  (Someone once told me -- I think it may have been you -- that "if the documentation and the code don't agree, they're both wrong" :) )

It's also a bit confusing that the _type_ of that parameter is given in the `@param` statement as `list of SubmissionSampleLocation`, whereas in the actual method header it is `SubmissionSolutionList`.  I actually don't think it would work to pass just any generic "list of SubmissionSampleLocation" to this parameter, since the actual type is an extension of `ArrayList<SubmissionSampleLocation>`.  (For example, a `LinkedList` is a "list", but you couldn't pass a `LinkedList<SubmissionSolutionList>` into the method...)

Reply:
will update the parameter, noting that above you indicated that users will likely ignore the comments.

Reply:
Hmm, note that SubmissionSolutionLis tis an ArrayList of SubmissionSampleLocation also can be used as a List of SubmissionSampleLocation(s).   I will update the comment perhaps to be more clear, but the SubmissionSolutionList is a list of SubmissionSampleLocation(s).